core/src/main/java/com/alibaba/druid/sql/dialect/oracle/parser/OracleStatementParser.java

```diff
@@ -1985,24 +1985,24 @@ private void parserParameters(List<SQLParameter> parameters, SQLObject parent) {
                     lexer.nextToken();
                     accept(Token.OF);
 
-                    name = this.exprParser.name();
+                    SQLName sqlName = this.exprParser.name();
 
```

Comment:
Take `TYPE Foursome IS TABLE OF VARCHAR2(15);` as example, the name should be the declared variable name `Foursome`, but since the the override, it became the `VARCHAR2`